lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -155,223 +182,188 @@
         return platform.tagPublicLibraryCode(f, name);
     };
 
-    var resolveLocaleBestFit = function (locale, defaultLocale) {
-        var resolvedLocale = platform.localeBestFitCache.get(locale);
+    const __resolveLocaleBestFit = new IntlCache();
+    const resolveLocaleBestFit = function (locale) {
+        var resolvedLocale = __resolveLocaleBestFit.get(locale);
         if (resolvedLocale === undefined) {
             resolvedLocale = platform.resolveLocaleBestFit(locale);
+            // TODO(jahorto): the fallback assumes that the provided locale is good, which seems bad
             if (resolvedLocale === null) {
-                if (!LANG_TAG_BASE_RE) {
-                    InitializeLangTagREs();
-                }
-                let match = platform.builtInRegexMatch(locale, LANG_TAG_BASE_RE);
-                resolvedLocale = match[1] + (match[2] ? ('-' + match[2]) : '') + (match[3] ? ('-' + match[3]) : '');
+                const match = platform.builtInRegexMatch(locale, LANG_TAG_BASE_RE);
```

Comment:
>LANG_TAG_BASE_RE [](start = 65, length = 16)

You must ensure that you use the pattern

```
                 if (!LANG_TAG_BASE_RE) {
                    InitializeLangTagREs();
                }
```

before using these regexes.

Reply:
I changed the regexes to always be initialized because we had that pattern so much.

Reply:
I'm not sure if it's the case that all the regexes are (or always need to be) initialized. If it's true that we will definitely initialize them during the Intl initialization, then always initializing them makes sense, otherwise there's potential to save runtime memory if the regexes are not needed for the operations actually used. It's a minor optimization but it's still a consideration to make.

Reply:
Perhaps we leave them uninitialized until one of the Initialize*() functions are called, and then initialize them in there? Theyre just used so often that it was becoming excessive to manually initialize them in each location